x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTruncProcessor.java

```diff
@@ -72,10 +69,10 @@ public static Object process(Object truncateTo, Object timestamp, ZoneId zoneId)
         if (timestamp instanceof ZonedDateTime == false
             && timestamp instanceof IntervalYearMonth == false
             && timestamp instanceof IntervalDayTime == false) {
-            throw new SqlIllegalArgumentException("A date/datetime/interval is required; received [{}]", timestamp);
+            throw new SqlIllegalArgumentException("A date/datetime/interval is required; received [{}]", timestamp); // verifier checked
```

Comment:
Shouldn't this also be `InvalidArgumentException`?

Reply:
The verifier checks this condition, so if it still gets here, it's a server error.